commit 6ea9514a454bc3cad3aeacca0865c709d4ecaa84 (refs/changes/68/1968/5)
Author: Chris Burroughs <chris.burroughs@joyent.com>
Date:   2017-05-26T20:23:02+00:00 (2 years, 4 months ago)
    
    TOOLS-1777 support configuring MG:jenkins-build output directory
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
    Reviewed by: Trent Mick <trent.mick@joyent.com>
    Approved by: Trent Mick <trent.mick@joyent.com>

diff --git a/tools/jenkins-build b/tools/jenkins-build
index 8d46b8b..ab1b434 100755
--- a/tools/jenkins-build
+++ b/tools/jenkins-build
@@ -6,7 +6,7 @@
 #
 
 #
-# Copyright (c) 2014, Joyent, Inc.
+# Copyright (c) 2017, Joyent, Inc.
 #
 
 #
@@ -31,6 +31,14 @@
 set -o errexit
 unset LD_LIBRARY_PATH   # ensure don't get Java's libs (see OS-703)
 
+
+#---- globals, config
+
+# The Manta directory that build artifacts are uploaded to.  The same
+# name is used for both public (under ~~/public) and private (under
+# ~~/stor) artifacts.
+MG_JENKINS_BUILDS_DIR=${MG_JENKINS_BUILDS_DIR:-builds}
+
 #
 # Refuse to build if /tmp is tmpfs until OS-5608 is fixed.
 # Try to umount if mounted, if that fails: blow up.
@@ -134,17 +142,18 @@ if [[ "$CLEAN_CACHE" == "true" ]]; then
     rm -rf $CACHE_DIR
 fi
 
-# Builds that are joyent-specific get uploaded to /stor/builds whereas all other
-# builds now go to /public/builds. So in either case, dependencies come from
-# /public/builds.
+# Builds that are joyent-specific get uploaded to a directory under
+# /stor whereas all other builds now go under /public.  The specific
+# directory is controlled by MG_JENKINS_BUILDS_DIR which defaults to
+# "builds". In either case, dependencies come from the public directory.
 if [[ ! -f targets.json ]]; then
     bash < targets.json.in | json > targets.json
 fi
 IS_PUBLIC=$(json < targets.json ${JOB_NAME}.public)
 if [[ ${IS_PUBLIC} == "true" ]]; then
-    JENKINS_OUTPUT_BASE="/public/builds"
+    JENKINS_OUTPUT_BASE="/public/${MG_JENKINS_BUILDS_DIR}"
 else
-    JENKINS_OUTPUT_BASE="/stor/builds"
+    JENKINS_OUTPUT_BASE="/stor/${MG_JENKINS_BUILDS_DIR}"
 fi
 
 if [[ -z ${JOYENT_BUILD} || ${JOYENT_BUILD} == "true" ]]; then
